### ISSUE TITLE
Editorconfig diagnostic severity configuration not properly respected for linked files

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#43371]
**Version Used**: Latest VS2019 16.7 dogfood build

**Steps to Reproduce**:

1. Extract [ClassLibrary38.zip](https://github.com/dotnet/roslyn/files/4482150/ClassLibrary38.zip) and double click `ClassLibrary38.sln` file inside it to open in VS.
2. Build `ClassLibrary1.csproj`

**Expected Behavior**:
No IDE0044 diagnostics

**Actual Behavior**:
IDE0044 is reported:
```
1>------ Rebuild All started: Project: ClassLibrary1, Configuration: Debug Any CPU ------
1>ClassLibrary38\Class1.cs(9,21,9,22): warning IDE0044: Make field readonly
```

Note that solution folder has an .editorconfig file with following contents, which should ensure `IDE0044` is not reported on any files within `ClassLibrary38`:
```ini
# IDE0044: Add readonly modifier

# Disable for entire folder and sub-folders
[*.{cs,vb}]
dotnet_diagnostic.IDE0044.severity = none

# Enable just for 'ClassLibrary1' and sub-folders
[ClassLibrary1/**/*.{cs,vb}]
dotnet_diagnostic.IDE0044.severity = warning
```

The fact that `Class1.cs` has been linked into `ClassLibrary1\ClassLibrary1.csproj` seems to be causing the compiler to incorrectly enable the diagnostic on `Class1.cs` when building `ClassLibrary1.csproj`.

Interestingly, the IDE live analysis behavior is correct - i.e. no IDE0044 is reported if you just open the file `Class1.cs` in IDE.

**Comment by mavasani** [comment]
Tag @sharwell - this is the issue that we are seeing when attempting to enforce some CodeStyle diagnostics as build warnings for `src\Workspaces` folder in Roslyn: https://github.com/dotnet/roslyn/blob/082d02d0616442b0370f28eecc981f8bcbec48d2/.editorconfig#L228

Internal utility files inside `src\Compilers` that are explicitly linked into `Workspaces.csproj` are being incorrectly flagged during build for rules enabled under the above editorconfig section. These diagnostics do not show up in the IDE live analysis for open files and are also not fixed as part of FixAll in project/solution application, which seems correct behavior.

**Comment by mavasani** [comment]
Tagging @agocke @jaredpar for visibility.

**Comment by davkean** [comment]
We're also hitting this. After this PR: https://github.com/dotnet/project-system/pull/6502. The command-line compiler is incorrectly applying this [.editorconfig](https://github.com/dotnet/project-system/blob/master/tests/.editorconfig) against [ManagedImageMonikers](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources/ManagedImageMonikers.cs)  for [this project](https://github.com/dotnet/project-system/blob/master/tests/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Microsoft.VisualStudio.ProjectSystem.IntegrationTests.csproj#L13), which results in no warnings/errors (incorrect) to be shown at command-line, but (correctly) shown inside Visual Studio:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	RS0016	Symbol 'Component.get' is not part of the declared API.	Microsoft.VisualStudio.ProjectSystem.IntegrationTests	C:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed\Resources\ManagedImageMonikers.cs	27	Active
Warning	RS0016	Symbol 'ComponentPrivate.get' is not part of the declared API.	Microsoft.VisualStudio.ProjectSystem.IntegrationTests	C:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed\Resources\ManagedImageMonikers.cs	28	Active
Warning	RS0016	Symbol 'ComponentWarning.get' is not part of the declared API.	Microsoft.VisualStudio.ProjectSystem.IntegrationTests	C:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed\Resources\ManagedImageMonikers.cs	29	Active
```

This resulted in this PR being green in CI, but red when you open Visual Studio.

**Comment by mavasani** [comment]
Tagging @chsienki

**Comment by sharwell** [comment]
@davkean prior to a fix, you can avoid the issue by applying the work around from 835c55b930f645109181dea980679df0bcbca9f9
